src/Stripe/Infrastructure/Requestor.cs

```diff
@@ -1,106 +1,182 @@
-﻿using System;
-using System.IO;
-using System.Net;
-using System.Text;
-
-namespace Stripe
-{
-    internal static class Requestor
-    {
-        public static string GetString(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "GET", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string PostString(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "POST", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string PostStringBearer(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "POST", apiKey, true);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        public static string Delete(string url, string apiKey = null)
-        {
-            var wr = GetWebRequest(url, "DELETE", apiKey);
-
-            return ExecuteWebRequest(wr);
-        }
-
-        internal static WebRequest GetWebRequest(string url, string method, string apiKey = null, bool useBearer = false)
-        {
-            apiKey = apiKey ?? StripeConfiguration.GetApiKey();
-
-            var request = (HttpWebRequest)WebRequest.Create(url);
-            request.Method = method;
-
-            if(!useBearer)
-                request.Headers.Add("Authorization", GetAuthorizationHeaderValue(apiKey));
-            else
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+
+#if NET40
+using System.Threading.Tasks;
+#endif
+
+namespace Stripe
+{
+    internal static class Requestor
+    {
+        public static string GetString(string url, string apiKey = null)
+        {
+            var wr = GetWebRequest(url, "GET", apiKey);

```

Comment:
Hey @paulirwin . Nice change! I was about to do this myself when I came across your pull request. One note that might help out @jaymedavis or any future developers in this code is to call the async call from the sync call to prevent code duplication.

For example:
```c#
return GetStringAsync(url, apiKey).GetAwaiter().GetResult();
```

Note that you can't just call ```task.Result``` or ```task.Wait()``` or exceptions will [bubble up as AggregateException rather than the original exception](http://stackoverflow.com/questions/17284517/is-task-result-the-same-as-getawaiter-getresult).

Thanks for doing this!